src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -159,7 +159,16 @@ internal static SafeSslContextHandle GetOrCreateSslContextHandle(SslAuthenticati
 
             if (sslAuthenticationOptions.IsClient)
             {
-                var key = new SslContextCacheKey(protocols, sslAuthenticationOptions.CertificateContext?.TargetCertificate.GetCertHash(HashAlgorithmName.SHA256));
+                byte[]? hash = null;
+
+                if (sslAuthenticationOptions.CertificateContext?.TargetCertificate is X509Certificate2 cert)
+                {
+                    // This is equivalent to cert.GetCertHash(HashAlgorithmName.Sha256), but this
+                    // way the code does not allocate a new byte[] with raw cert contents.every time
+                    hash = SHA256.HashData(cert.RawDataMemory.Span);
+                }
+
+                var key = new SslContextCacheKey(protocols, hash);
```

Comment:
@bartonjs is the `X509Certificate.GetCertHash(...)` behavior by design? looks like something that could potentially help in multiple places.

Reply:
(Not Jeremy, but...) I don't think so. We can probably make some improvements here.

Reply:
Since it's using certPal.RawData instead of this.RawData, I think it's trying to avoid the allocation; but clearly some PALs are caching that and others are not, because we started off with a weak PAL contract and haven't ever cleaned it up.

In this case, it looks like just switching from certPal.RawData to this.RawDataMemory.Span will mean we know exactly what caching is involved.

Reply:
With https://github.com/dotnet/runtime/pull/103828 merged I am not sure this is needed anymore.